test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java

```diff
@@ -40,6 +45,27 @@ public final void testFromXContent() throws IOException {
             .test();
     }
 
+    /**
+     * Calls {@link ToXContent#toXContent} on many threads and verifies that they produce the same result.
+     */
+    public final void testConcurrentToXContent() throws IOException, InterruptedException, ExecutionException {
+        XContentType xContentType = randomFrom(XContentType.values());
+        T testInstance = createXContextTestInstance(xContentType);
+        ToXContent.Params params = new ToXContent.MapParams(singletonMap(RestSearchAction.TYPED_KEYS_PARAM, "true"));
+        boolean humanReadable = randomBoolean();
+        BytesRef firstTimeBytes = toXContent(testInstance, xContentType, params, humanReadable).toBytesRef();
+
+        concurrentTest(() -> {
+            try {
+                for (int r = 0; r < 500; r++) {
```

Comment:
I would randomize these and in concurrent tests as well, so we vary it.

Reply:
I think randomizing it lower just makes it a worse test. We could randomly make it higher, but that'd make the test run times even more unpredictable. Which is fine, I guess. Do you think something like `between(500, 5000)` or something?

Reply:
Here is the reason for my comment. The only explanation I could come up with why it was 500 here is because it was reliably failing in this particular case on Nik's machine at that moment. It is pretty obvious that the higher the numbers the more chances for this test to reliably fail but it would also increase the duration of this test. So, I thought it might be better to just add some variation so sometimes it really beats crap out of it but generally doesn't take too much time, so we can cover a variety of hardware and failure scenarios. I don't have any strong feeling about it, so I think it might be fine to just leave it as is.